media/public/include/IMediaPipeline.h

```diff
@@ -1183,6 +1183,30 @@ class IMediaPipeline
      */
     virtual bool getPosition(int64_t &position) = 0;
 
+    /**
+     * @brief Sets the "Immediate Output" property for this source.
+     *
+     * This method is sychronous, it sets the "Immediate Output" property
+     *
+     * @param[in] sourceId  : The source id. Value should be set to the MediaSource.id returned after attachSource()
+     * @param[in] immediateOutput : Set immediate output mode on the sink
+     *
+     * @retval true on success.
+     */
+    virtual bool setImmediateOutput(int32_t sourceId, bool immediateOutput) = 0;
+
+    /**
+     * @brief Sets the "Immediate Output" property for this source.
+     *
+     * This method is sychronous, it sets the "Immediate Output" property
+     *
+     * @param[in] sourceId  : The source id. Value should be set to the MediaSource.id returned after attachSource()
+     * @param[in] immediateOutput : Set immediate output mode on the sink
```

Comment:
I think this should be a param[out] to get the immediate output mode on the sink